datahub-web-react/src/app/glossary/GlossaryBrowser/NodeItem.tsx

```diff
@@ -91,6 +91,13 @@ function NodeItem(props: Props) {
         }
     }, [refreshBrowser]);
 
+    useEffect(() => {
+        if (updatedUrns.includes(node.urn)) {
+            refetch();
+            setUpdatedUrns(updatedUrns.filter((urn) => urn !== node.urn));
```

Comment:
So after an update the leaf component is responsible for removing itself from the updated Urns list?

Minor: but maybe we consider calling this urnsToUpdate to make the transaction that's happening here clearer

Reply:
Also, is there any risk in having a single array here?

Instead of urnsToUpdate and updatedUrns as separate structures

Reply:
ooo i definitely like having `urnsToUpdate` as a name much more here. and yeah having a singular array i think is fine as long as we're smart about only updating state when se need to (like after the entity that needs to update refetches and only if it's already in the array).

Will update the name!